tests/actions/test.transaction.js

```diff
@@ -99,4 +106,131 @@ describe('transactionActions', function() {
 
   });
 
+  describe('processPayment', function() {
+
+    function processPayment({fetch, productId,
+                             client=helpers.FakeBraintreeClient,
+                             payMethod=fakeCreditCard}) {
+      var deferredAction = transactionActions.processPayment({
+        productId: productId,
+        payMethod: payMethod,
+        braintreeToken: 'braintree-token',
+        fetch: fetch,
+        BraintreeClient: client,
+      });
+
+      helpers.doApiAction(deferredAction, dispatchSpy);
+    }
+
+    it('should create a subscription for recurring products', function() {
+      var fetch = helpers.fakeFetch();
+      processPayment({fetch: fetch, productId: 'mozilla-concrete-brick'});
+      assert.equal(fetch.firstCall.args[0].url, '/braintree/subscriptions/');
+    });
+
+    it('should create a sale for one-off products', function() {
+      var fetch = helpers.fakeFetch();
+      processPayment({fetch: fetch, productId: 'mozilla-foundation-donation'});
```

Comment:
Yeah, I was thinking about that too. It's a bad pattern that we keep building on. Filed: https://github.com/mozilla/payments-ui/issues/353

It's a little tricky for a couple reasons:
- lots of code uses `products.get(id)` rather than accept a complete object
- the config is in flux so it's kind of nice to use real-ish configured products